src/Psalm/Type/UnionTrait.php

```diff
@@ -1016,6 +1017,25 @@ public function isString(bool $check_templates = false): bool
         ) === count($this->types);
     }
 
+    /**
+     * @psalm-mutation-free
+     * @return bool true if this is a string
+     */
+    public function isNonEmptyString(bool $check_templates = false): bool
+    {
+        return count(
+            array_filter(
+                $this->types,
+                static fn($type): bool => $type instanceof TNonEmptyString
+                    || ($type instanceof TLiteralString && $type->value)
```

Comment:
The way you checked, you inferred a non-falsy-string: '0' is non empty but it will return false

You need to explicitly check `!== ''` to check for non-emptyness

Reply:
D'oh, well spotted. Now updated.